test/browsing.js

```diff
@@ -229,6 +229,107 @@ describe('Browsing', () => {
+            })
+        ));
+    }).timeout(20000);
+
+    it('fetches partial content starting at 0', () => {
+      const content = `hello world, on ${Math.round(Math.random() * 100000)}`;
+      const numberOfBytes = content.length - 1;
+
+      const bytesToRetrieve = 7;
+      const startByte = 0;
+
+      return createRandomDomain(content, '/streaming.mp4')
+        .then((domain) => createAnonTestApp()
+          .then((app) => app.webFetch(`safe://${domain}/streaming.mp4`,
+                    { range: { start: startByte, end: startByte + bytesToRetrieve } })
+            .then((data) => {
+              should(data.body.toString()).equal('hello w');
+              should(data.body.toString()).equal(
+                          content.substring(startByte, startByte + bytesToRetrieve));
+              should(data.headers['Content-Range']).equal(`bytes ${startByte}-${startByte + (bytesToRetrieve)}/${numberOfBytes}`);
+              should(data.headers['Content-Length']).equal(bytesToRetrieve);
+            })
+        ));
+    }).timeout(20000);
+
+    it('fetch full length with range', () => {
+      const content = `hello world, on ${Math.round(Math.random() * 100000)}`;
+      const numberOfBytes = content.length - 1;
+      return createRandomDomain(content, '/streaming.mp4')
+        .then((domain) => createAnonTestApp()
+          .then((app) => app.webFetch(`safe://${domain}/streaming.mp4`, { range: { start: 0, end: numberOfBytes } })
+            .then((data) => {
+              should(data.body.toString()).equal(content);
+              should(data.headers['Content-Range']).equal(`bytes 0-${numberOfBytes}/${numberOfBytes}`);
+              should(data.headers['Content-Length']).equal(content.length);
+            })
+        ));
+    }).timeout(20000);
+
+    it('without range end param', () => {
+      const content = `hello world, on ${Math.round(Math.random() * 100000)}`;
+      const startByte = 4;
+      const numberOfBytes = content.length - 1;
+
+      return createRandomDomain(content, '/streaming.mp4')
+        .then((domain) => createAnonTestApp()
+          .then((app) => app.webFetch(`safe://${domain}/streaming.mp4`, { range: { start: startByte } })
+            .then((data) => {
+              should(data.body.toString()).equal(content.substring(startByte));
+              should(data.headers['Content-Range']).equal(`bytes ${startByte}-${numberOfBytes}/${numberOfBytes}`);
+              should(data.headers['Content-Length']).equal(content.length - startByte);
+            })
+        ));
+    }).timeout(20000);
+
+    it('range end beyond data length', () => {
+      const content = `hello world, on ${Math.round(Math.random() * 100000)}`;
+      return createRandomDomain(content, '/streaming.mp4')
+        .then((domain) => createAnonTestApp()
+          .then((app) => should(app.webFetch(`safe://${domain}/streaming.mp4`, { range: { start: 0, end: content.length + 1 } }))
```

Comment:
Perhaps we can pass the `end ` with `content.length` rather than `content.length + 1` to make sure we test the limit.

Reply:
You mean in the HTTP req?